### ISSUE TITLE
Protocol and hostname not removed when tracking

### ISSUE THREAD

**Issue by aldo-roman** [umami-software/umami#2336]
## Expected behavior

A. Request from web `https://mydomain.com/path` recorded as visit to `/path`
B. Request from mobile app `capacitor://localhost/path` recorded as visit to `/path`

## Actual behavior

A. Request from web `https://mydomain.com/path` recorded as visit to `/path`
B. Request from mobile app `capacitor://localhost/path` recorded as visit to `capacitor://localhost/path`

## Notes
This occurs when tracking web apps using Capacitor. The visited URL is not `https://<domain>/<path>` but `capacitor://localhost/<path>`.
This makes it difficult to aggregate visits from native apps to pages.

### Database

PostgreSQL

### Which browser are you using? (if relevant)

Safari web view

**Comment by aldo-roman** [comment]
Probably related to this function:
https://github.com/umami-software/umami/blob/fd5d5c692e09fa5b756d0ca7b2a8c104e1c80b4a/src/tracker/index.js#L45-L50

```diff
 const getPath = url => { 
-   if (url.substring(0, 4) === 'http') { 
-     return '/' + url.split('/').splice(3).join('/'); 
-   } 
-   return url; 
+   return new URL(url).pathname;
 }; 
```

**Comment by mikecao** [comment]
This will not work if the path is not a full URL like `/`. Perhaps we can check for the `://` and allow for any protocol.

**Comment by aldo-roman** [comment]
@mikecao can you expand on when do these cases happen?

`/` is not a valid URL. In such case, we could just return the whole url parameter as it was done.

```ts
const getPath = url => {
  try {
    return new URL(url).pathname;
  } catch {
    return url;
  }
};
```

**Comment by mikecao** [comment]
When you go from one page to another within the same website, the referrer is set to just the pathname, not the entire URL. But I think your updated solution would work.

**Comment by aldo-roman** [comment]
Got it
Please see PR #2337

**Comment by mikecao** [comment]
Fixed in v2.8.0.
